From 4cbae57cef65df8705b77cf882372f835a42d842 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 26 Oct 2012 17:59:50 +0100
Subject: [PATCH 053/278] assdec: fix qsort() callback signature

This changes the event_cmp() function to the correct signature,
avoiding an ugly cast.

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 libavformat/assdec.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libavformat/assdec.c b/libavformat/assdec.c
index 3d39845..0041ca4 100644
--- a/libavformat/assdec.c
+++ b/libavformat/assdec.c
@@ -68,8 +68,9 @@ static int64_t get_pts(const uint8_t *p)
     return sec*100+hsec;
 }
 
-static int event_cmp(uint8_t **a, uint8_t **b)
+static int event_cmp(const void *_a, const void *_b)
 {
+    const uint8_t *const *a = _a, *const *b = _b;
     return get_pts(*a) - get_pts(*b);
 }
 
@@ -131,7 +132,7 @@ static int read_header(AVFormatContext *s)
         p++;
     }
 
-    qsort(ass->event, ass->event_count, sizeof(*ass->event), (void*)event_cmp);
+    qsort(ass->event, ass->event_count, sizeof(*ass->event), event_cmp);
 
     return 0;
 
-- 
1.7.9.5

